server/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverResponse.java

```diff
@@ -114,34 +115,60 @@ public boolean isRolledOver() {
 
     @Override
     public void readFrom(StreamInput in) throws IOException {
-        super.readFrom(in);
-        oldIndex = in.readString();
-        newIndex = in.readString();
-        int conditionSize = in.readVInt();
-        conditionStatus = new HashMap<>(conditionSize);
-        for (int i = 0; i < conditionSize; i++) {
-            conditionStatus.put(in.readString(), in.readBoolean());
+        if (in.getVersion().onOrAfter(Version.V_7_0_0_alpha1)) {
+            super.readFrom(in);
+            oldIndex = in.readString();
+            newIndex = in.readString();
+            int conditionSize = in.readVInt();
+            conditionStatus = new HashMap<>(conditionSize);
+            for (int i = 0; i < conditionSize; i++) {
+                conditionStatus.put(in.readString(), in.readBoolean());
+            }
+            dryRun = in.readBoolean();
+            rolledOver = in.readBoolean();
+            readShardsAcknowledged(in);
+        } else {
+            oldIndex = in.readString();
```

Comment:
here we are lucky that super reads only the acknowledged flag and we get away with effectively not calling super, it's a tricky one ;)

I would probably make a method with this shared code block

Reply:
I'm not sure I understand which code block to share. Do you mean the part in super() that writes the boolean flag or the common read parts of the two branches here? In the later case this isn't completely possible since in the "old" serialization order the acknowledged field gets read as second-to-last field. Maybe I didn't understand this correctly.

Reply:
arg I hadn't seen the acknowledged flag, nevermind